src/main/java/org/embulk/filter/encrypt/EncryptFilterPlugin.java

```diff
@@ -106,12 +106,58 @@ public String toString()
         }
     }
 
+    public enum Encoder
+    {
+        BASE64("base64", BaseEncoding.base64()),
+        HEX("hex", BaseEncoding.base16());
+
+        private final BaseEncoding encoding;
+        private final String name;
+
+        Encoder(String name, BaseEncoding encoding)
+        {
+            this.name = name;
+            this.encoding = encoding;
+        }
+
+        public String encode(byte[] bytes)
+        {
+            return encoding.encode(bytes);
+        }
+
+        @JsonCreator
+        public static Encoder fromName(String name)
+        {
+            EnumSet<Encoder> encoders = EnumSet.allOf(Encoder.class);
+            for (Encoder encoder : encoders) {
+                if (encoder.name.equals(name)) {
+                    return encoder;
+                }
+            }
+            throw new ConfigException(
+                    format("Unsupported encoding format '%s'. Supported encoding formats are %s.",
+                            name,
+                            join(encoders, ", ")));
+        }
+
+        @JsonValue
+        @Override
+        public String toString()
+        {
+            return name;
+        }
+    }
+
     public interface PluginTask
             extends Task
     {
         @Config("algorithm")
         public Algorithm getAlgorithm();
 
+        @Config("format")
```

Comment:
The name `format` sounds a bit different from what it does. It is a kind of encoding, not just formatting.

What do you think `output_encoding` or like that?

Reply:
I do agree